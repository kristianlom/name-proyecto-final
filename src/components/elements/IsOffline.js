import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

const IsOffline = (props) => {
  const [online, setOnline] = useState(navigator ? navigator.onLine : true);

  const goOnline = () => {
    setOnline(true);
  };

  const goOffline = () => {
    setOnline(false);
  };

  useEffect(() => {
    if (!window) return;
    window.addEventListener('online', goOnline);
    window.addEventListener('offline', goOffline);

    // eslint-disable-next-line consistent-return
    return () => {
      window.removeEventListener('online', goOnline);
      window.removeEventListener('offline', goOffline);
    };
  }, [online]);

  if (online) {
    return null;
  }

  return (
    <span className="header__offline">
      {props.children}
    </span>
  );
};

IsOffline.propTypes = {
  children: PropTypes.node.isRequired,
};

export default IsOffline;
