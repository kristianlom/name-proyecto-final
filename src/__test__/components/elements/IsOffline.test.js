import React from 'react';
import { mount } from 'enzyme';
import IsOffline from '../../../components/elements/IsOffline';

describe('Testing render <IsOffline /> Component', () => {
  let wrapperComponent;
  beforeEach(() => {
    wrapperComponent = mount(<IsOffline />);
  });
  // ---Snapshot:
  it('create snapshot', () => {
    expect(wrapperComponent.find(IsOffline)).toMatchSnapshot();
  });
  //  ---Rendering:
  it('component renders correctly', () => {
    expect(wrapperComponent.find(IsOffline).length).toEqual(1);
  });
});
