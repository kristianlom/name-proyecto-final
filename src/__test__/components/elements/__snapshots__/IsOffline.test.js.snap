// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing render <IsOffline /> Component create snapshot 1`] = `ReactWrapper {}`;
